python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -140,10 +141,51 @@ def execute_action(self, data):
                 return
 
         project_entity, entities = self._get_entities_from_payload(data)
+        config_entity = data["pc"]
+        command_name = data["name"]
+
+        # We have to do a couple of things here. The first is that we ALWAYS
+        # stick to the non-legacy code path for the __core_info and __upgrade_check
+        # commands. Neither of these require a bootstrap to occur, and are therefore
+        # fast when run through the modern code path. In addition, the implementation
+        # of these "special" project-level commands already outputs markdown syntax,
+        # which is preferable to the HTML we would get from the tank command.
+        #
+        # The second bit is that we can identify commands that need to be run in
+        # legacy mode by whether the config entity dictionary we got from the client
+        # contains a "_legacy_config_root" key that was shoved into it when the
+        # get_actions method ran through its own legacy code path. In that case, we
+        # shell out to the tank command by way of our process_manager object instead
+        # of using the modern code path.
+        if command_name not in ["__core_info", "__upgrade_check"]:
+            if "_legacy_config_root" in config_entity:
+                (out, err, retcode) = self.process_manager.execute_toolkit_command(
+                    config_entity["_legacy_config_root"],
+                    "shotgun_run_action",
+                    [data["name"], entities[0]["type"], str(entities[0]["id"])],
+                )
+
+                # Sanitize the output. By going the legacy route here, we're going
+                # to end up with HTML in the output from the tank command. We need
+                # to filter that and sanitize anything we keep, because the client
+                # is going to believe that we're sending v2-style output, which is
+                # taken and displayed as is, and is assumed to be markdown and not
+                # HTML.
+                out = self._legacy_sanitize_output(out)
+
+                self.host.reply(
+                    dict(
+                        retcode=retcode,
+                        err=err,
+                        out=out,
+                    )
+                )
+
+                return
```

Comment:
My reply of a semi-nitpick is that I equally dislike HUGE if/else blocks that get the same result as a return. :/